### ISSUE TITLE
Reproduce the results in the paper

### ISSUE THREAD

**Issue by lintian-a** [cwmok/Fast-Symmetric-Diffeomorphic-Image-Registration-with-Convolutional-Neural-Networks#17]
Hi, thanks for this awesome work on the topic of advocating symmetric registration. The idea of utilizing a halfway geodesic trajectory is very interesting. I am currently trying to reproduce the results in the paper and I have the following questions regarding the test result:

1) Regarding the dataset split for train, validation, and testing. In the Train_sym_neurite_oasis, the validation set is the oasis list from 255 to 260. This is slightly different from the description in the paper. Do you mind sharing the case ids for the test split you used in the paper?

2) In https://github.com/adalca/medical-datasets/blob/master/neurite-oasis.md, the whole dataset seems to contain 414 subjects. This is different from the paper. Is this the dataset I should use to reproduce the result? 

3) In the paper, it seems that there is some preprocessing conducted on the oasis dataset. Is the dataset downloaded from https://github.com/adalca/medical-datasets/blob/master/neurite-oasis.md already preprocessed the same way as in the paper? Does one need to apply extra preprocessing based on the oasis downloaded from the above link?

**Comment by cwmok** [comment]
Hi @lintian-a,

First of all, congratulation on your CVPR's paper this year! Your GradICON is very solid and the result is promising.

I will answer your questions below:

1. The "Train_sym_neurite_oasis.py" is to provide a toy example to demonstrate how SYMNet can be trained on the existing dataset. In our paper, we strictly follow the dataset split as written in our paper, i.e., train/validation/test: 255/20/150, order by the case ID (ascending order). In the validation phase of "Train_sym_neurite_oasis.py", we measure the pairwise registration result, but we measure the atlas registration result (register all validation scans to the 1st validation scan) during the experiments in our paper. Therefore, we reduce the number of validation scans in our toy example in order to speed up the training. 

2. Yes, you are correct. Our paper was actually published earlier than adalca's preprocessed dataset.  In the paper, we used the official raw data downloaded from https://www.oasis-brains.org/#data via FTP. And preprocessed it with Freesurfer, visual check and cropped it from 256^3 to 144x192x160. Finally, we obtained 425 T1-weighted brain MRI, which is different from the 414 scans in adalca's preprocessed dataset. An example of our preprocessed data is provided at https://github.com/cwmok/Fast-Symmetric-Diffeomorphic-Image-Registration-with-Convolutional-Neural-Networks/tree/master/Data.

3. I think we are using a very similar preprocessing pipeline as we both use Freesurfer to do so. Yet, there are slight differences such as visual check, the orientation of the scans and the number of scans included in the dataset. If you would like to pre-process the raw data on your own, you may follow the [junyuchen245](https://github.com/junyuchen245/TransMorph_Transformer_for_Medical_Image_Registration/commits?author=junyuchen245)'s guideline here: https://github.com/junyuchen245/TransMorph_Transformer_for_Medical_Image_Registration/blob/main/PreprocessingMRI.md

4. If you want to strictly reproduce the results (with 425 scans) in the paper, you need to download the OASIS raw data and follow the preprocessing steps above. If you want to reproduce the results with adalca's processed dataset, you might try our toy example and adjust the hyperparameter of SYMNet.

**Comment by lintian-a** [comment]
Hi Tony,

Thank you very much! 

And thank you for the detailed answers. I still have this one question. I saw there are 5 types of oasis datasets on https://www.oasis-brains.org/#data.  Do you mind pointing me to the one you used in the paper?  I just need to get the id list of the test set so that I can rule out the trainset from adalca's processed dataset.

**Comment by cwmok** [comment]
Hi Lintian,

We only use the "OASIS-1: Cross-sectional MRI Data in Young, Middle Aged, Nondemented and Demented Older Adults" one. I downloaded the dataset in 2018, which contained only 425 T1-weighted brain MRI. I believe they have updated the dataset since 2019.

**Comment by lintian-a** [comment]
Got it. 

Since the current OASIS-1 only contains 416 subjects, it would be very difficult to rule out the train set and validation set from the OASIS-1 id list based on the rule:

>  "In our paper, we strictly follow the dataset split as written in our paper, i.e., train/validation/test: 255/20/150, order by the case ID (ascending order)."

 Is it possible that you can share the smallest case id for the test set used in the paper? With this information, I can just use all the cases with greater case ids as the test set.

**Comment by cwmok** [comment]
Update: In short, for the test set, the smallest case id for the test set will be OAS1_0290_MR1 in disc 7.

----------------------------------------------------
We use all disc (1-11) in the OASIS-1 dataset. If you count the total number of MR sessions in the screenshot, you will find out there are 425 items.
![image](https://user-images.githubusercontent.com/63316551/222880852-e576312c-ed95-4b58-a901-0273105be707.png)

The smallest case id will be OAS1_0001_MR1.
![image](https://user-images.githubusercontent.com/63316551/222880862-d16f56a6-812c-4074-8994-b01cae3263eb.png)

The largest case id will be OAS1_0457_MR1.
![image](https://user-images.githubusercontent.com/63316551/222880878-0799875a-1a38-476b-b451-c98c7e614687.png)


For the test set, the smallest case id for the test set will be OAS1_0290_MR1 in disc 7.

**Comment by lintian-a** [comment]
Got it. Thank you very much for the patient answers! This helps a lot!

One thing to note,  some cases might have been removed from the dataset since the version you used. The oasis-1 from https://www.oasis-brains.org/#data contains 416 T1 cases and the oasis from https://github.com/adalca/medical-datasets/blob/master/neurite-oasis.md contains 414 cases. Both of them end with 0457 being the greatest id.
